### PR TITLE
Migrated the style guide from google docs to markdown

### PR DESCRIPTION
**Pull request by MichaelLyst**
@wolffan Do you want to review this before we publish it to the world? :-)

### CODE REVIEW

**Comment by MichaelLyst** [comment]
@wolffan o_O

File: style-guide.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+This not only gives you better type safety at compile time, but can avoid some nasty gotchas.
+
+Favour initialisers for classes expressed as class methods. For example:
+
+	+(instancetype)classFactory
+	{
+ 		return [[self alloc] init];
+	}
+
+Caveat: beware of calling allow with the specific class name (.e.g [MyClass alloc] init]). Doing so means, then we won’t be able to create subclasses.
```
Comment (wolffan) [inline style-guide.md]: there's a typo in allow ;)

File: style-guide.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	__weak typeof(self) weakSelf = self;
+
+And then transform it to a strong one inside of the block:
+
+	__strong typeof(self) strongSelf = weakSelf;
+
+then use strongSelf.whatever in your block. Not using a strong reference will cause a compiler error.
+
+### ! conditional statements
+In order to prepare for our migration to Swift, all future conditionals using the ! operator are to be avoided unless they act upon a true boolean ([https://developer.apple.com/library/prerelease/mac/documentation/Swift/Conceptual/Swift_Programming_Language/BasicOperators.html](developer.apple.com/library/prerelease/mac/documentation/Swift/Conceptual/Swift_Programming_Language/BasicOperators.html)).
```
Comment (wolffan) [inline style-guide.md:130]: is there a formatting issue here? It's all bold. (maybe its Github being nasty)

File: style-guide.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+All properties should be nonatomic until they cannot be (for speed)
+
+	@property (nonatomic, copy, readonly) NSString *name;
+
+### Copy properties for mutable types
+All NSString, NSDictionary and NSArray properties should be copy, not strong (to prevent being handed an NSMutable version which could then change)
+
+	@property (nonatomic, copy, readonly) NSString *name;
+
+### When is is not good to use properties
```
Comment (wolffan) [inline style-guide.md]: when is it not good ...? (typo?)

**Comment by wolffan** [comment]
Found a couple of small typos and I think we may have some code formatting issues.

Also not sure how the result will look like, without wanting to be an ass, I'd check the NYTimes https://github.com/NYTimes/objective-c-style-guide#conditionals that is the one I kinda "stole" back then for mine.

**Pull request merged**